addons/account/models/account_analytic_account.py

```diff
@@ -18,26 +18,40 @@ class AccountAnalyticAccount(models.Model):
     @api.depends('line_ids')
     def _compute_invoice_count(self):
         sale_types = self.env['account.move'].get_sale_types(include_receipts=True)
-        domain = [
-            ('move_line_id.move_id.move_type', 'in', sale_types),
-            ('account_id', 'in', self.ids)
-        ]
-        groups = self.env['account.analytic.line']._read_group(domain, ['move_line_id.move_id:count_distinct'], ['account_id'])
-        moves_count_mapping = dict((g['account_id'][0], g['account_id_count']) for g in groups)
+
+        query = self.env['account.move.line']._search([
+            ('move_id.state', '=', 'posted'),
+            ('move_id.move_type', 'in', sale_types),
+        ])
+        query.add_where('account_move_line.analytic_distribution ?| array[%s]', [str(account_id) for account_id in self.ids])
+
+        query.order = None
+        query_string, query_param = query.select('jsonb_object_keys(account_move_line.analytic_distribution) as distribution', 'COUNT(DISTINCT(move_id)) as moves')
```

Comment:
* the aliases could be more meaningful
* don't be afraid of having more lines ;)

```suggestion
        query_string, query_param = query.select(
            'jsonb_object_keys(account_move_line.analytic_distribution) AS account',
            'COUNT(DISTINCT(move_id)) AS move_count',
        )
```